src/python/grpcio/grpc/_cython/_cygrpc/metadata.pyx.pxi

```diff
@@ -0,0 +1,64 @@
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+
+_Metadatum = collections.namedtuple('_Metadatum', ('key', 'value',))
+
+
+cdef void _c_metadata(metadata, size_t *c_count, grpc_metadata **c_metadata):
+ import logging
+ try:
+  if metadata is None:
+    c_count[0] = 0
+    c_metadata[0] = NULL
+  else:
+    metadatum_count = len(metadata)
+    if metadatum_count == 0:
+      c_count[0] = 0
```

Comment:
I assume this  (```[0]```) is a workaround an issue with the * operator?  Can you link to a bug if it is a known cython issue?

Reply:
I feel similarly but I don't think it's a workaround. I think it might actually be the way Cython code is supposed to be written; Cython flatly suggests doing so with the message `a starred assignment target must be in a list or tuple - maybe you meant to use an index assignment: var[0] = ...` when I try to write it with the `*` operator.

Reply:
I just looked at the docs, you are right this is how Cython wants to dereference things...